File: Recipe_App/student_management.py
import os

FILENAME = "students.txt"

def add_student():
    with open(FILENAME, "a") as f:
        roll = input("Enter Roll Number: ")
        name = input("Enter Name: ")
        marks = input("Enter Marks: ")
        f.write(f"{roll},{name},{marks}\n")
    print("Student added successfully!\n")

def display_students():
    print("\n--- Student Records ---")
    try:
        with open(FILENAME, "r") as f:
            for line in f:
                roll, name, marks = line.strip().split(",")
                print(f"Roll: {roll}, Name: {name}, Marks: {marks}")
    except FileNotFoundError:
        print("No records found.")
    print()

def search_student():
    roll_no = input("Enter Roll Number to Search: ")
    found = False
    try:
        with open(FILENAME, "r") as f:
            for line in f:
                roll, name, marks = line.strip().split(",")
                if roll == roll_no:
                    print(f"Record Found: Roll: {roll}, Name: {name}, Marks: {marks}")
                    found = True
                    break
        if not found:
            print("Student not found.\n")
    except FileNotFoundError:
        print("No records found.\n")

def update_student():
    roll_no = input("Enter Roll Number to Update: ")
    updated = False
    try:
        lines = []
        with open(FILENAME, "r") as f:
            lines = f.readlines()
        with open(FILENAME, "w") as f:
            for line in lines:
                roll, name, marks = line.strip().split(",")
                if roll == roll_no:
                    print("Enter new details:")
                    name = input("New Name: ")
                    marks = input("New Marks: ")
                    f.write(f"{roll},{name},{marks}\n")
                    updated = True
                else:
                    f.write(line)
        if updated:
            print("Student record updated.\n")
        else:
            print("Student not found.\n")
    except FileNotFoundError:
        print("No records found.\n")

def delete_student():
    roll_no = input("Enter Roll Number to Delete: ")
    deleted = False
    try:
        lines = []
        with open(FILENAME, "r") as f:
            lines = f.readlines()
        with open(FILENAME, "w") as f:
            for line in lines:
                roll, name, marks = line.strip().split(",")
                if roll != roll_no:
                    f.write(line)
                else:
                    deleted = True
        if deleted:
            print("Student record deleted.\n")
        else:
            print("Student not found.\n")
    except FileNotFoundError:
        print("No records found.\n")

def menu():
    while True:
        print("\n==== Student Record Management ====")
        print("1. Add Student")
        print("2. Display All Students")
        print("3. Search Student")
        print("4. Update Student")
        print("5. Delete Student")
        print("6. Exit")
        choice = input("Enter your choice: ")

        if choice == '1':
            add_student()
        elif choice == '2':
            display_students()
        elif choice == '3':
            search_student()
        elif choice == '4':
            update_student()
        elif choice == '5':
            delete_student()
        elif choice == '6':
            print("Exiting program.")
            break
        else:
            print("Invalid choice! Please try again.")

# Run the program
menu()

File: Recipe_App/student_management.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

struct Student {
    int roll;
    char name[50];
    float marks;
};

void addStudent() {
    FILE *fptr = fopen("students.txt", "a");
    struct Student s;

    printf("Enter Roll No: ");
    scanf("%d", &s.roll);
    printf("Enter Name: ");
    scanf(" %[^\n]", s.name);
    printf("Enter Marks: ");
    scanf("%f", &s.marks);

    fprintf(fptr, "%d,%s,%.2f\n", s.roll, s.name, s.marks);
    fclose(fptr);

    printf("Student added successfully!\n");
}

void displayStudents() {
    FILE *fptr = fopen("students.txt", "r");
    struct Student s;

    if (fptr == NULL) {
        printf("No records found.\n");
        return;
    }

    printf("\n--- Student Records ---\n");
    while (fscanf(fptr, "%d,%49[^,],%f\n", &s.roll, s.name, &s.marks) != EOF) {
        printf("Roll: %d, Name: %s, Marks: %.2f\n", s.roll, s.name, s.marks);
    }

    fclose(fptr);
}

void searchStudent() {
    int rollNo, found = 0;
    struct Student s;
    FILE *fptr = fopen("students.txt", "r");

    if (fptr == NULL) {
        printf("No records found.\n");
        return;
    }

    printf("Enter Roll Number to search: ");
    scanf("%d", &rollNo);

    while (fscanf(fptr, "%d,%49[^,],%f\n", &s.roll, s.name, &s.marks) != EOF) {
        if (s.roll == rollNo) {
            printf("Record Found: Roll: %d, Name: %s, Marks: %.2f\n", s.roll, s.name, s.marks);
            found = 1;
            break;
        }
    }

    if (!found) {
        printf("Student not found.\n");
    }

    fclose(fptr);
}

void deleteStudent() {
    int rollNo, found = 0;
    struct Student s;
    FILE *fptr = fopen("students.txt", "r");
    FILE *temp = fopen("temp.txt", "w");

    if (fptr == NULL) {
        printf("No records found.\n");
        return;
    }

    printf("Enter Roll Number to delete: ");
    scanf("%d", &rollNo);

    while (fscanf(fptr, "%d,%49[^,],%f\n", &s.roll, s.name, &s.marks) != EOF) {
        if (s.roll != rollNo) {
            fprintf(temp, "%d,%s,%.2f\n", s.roll, s.name, s.marks);
        } else {
            found = 1;
        }
    }

    fclose(fptr);
    fclose(temp);
    remove("students.txt");
    rename("temp.txt", "students.txt");

    if (found) {
        printf("Student record deleted.\n");
    } else {
        printf("Student not found.\n");
    }
}

int main() {
    int choice;
    while (1) {
        printf("\n===== Student Record Management =====\n");
        printf("1. Add Student\n");
        printf("2. Display Students\n");
        printf("3. Search Student\n");
        printf("4. Delete Student\n");
        printf("5. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1: addStudent(); break;
            case 2: displayStudents(); break;
            case 3: searchStudent(); break;
            case 4: deleteStudent(); break;
            case 5: exit(0);
            default: printf("Invalid choice!\n");
        }
    }

    return 0;
}
